File: common/tokens.ts
export type RefreshToken = string;
export type AccessToken = string;

export interface Tokens {
    accessToken: AccessToken;
    refreshToken: RefreshToken;
}

export interface JwtPayload {
    createdAt: Date;
    expiresAt: Date;
    username: string;
}

export interface RawPayload {
    iat: number;
    exp: number;
    username: string;
}

File: client_leger/lib/interfaces/tokens.dart
import 'package:dart_jsonwebtoken/dart_jsonwebtoken.dart';

class TokenPayload {
  final String username;
  final DateTime expiresAt;
  final DateTime createdAt;

  TokenPayload(this.username, this.expiresAt, this.createdAt);

  bool isExpired() {
    return DateTime.now().isAfter(expiresAt);
  }
}

class RawTokens {
  final String accessToken;
  final String refreshToken;

  RawTokens(this.accessToken, this.refreshToken);
}

class Tokens {
  final JWT? accessToken;
  final JWT? refreshToken;

  Tokens(this.accessToken, this.refreshToken);
}
